scipy/stats/tests/test_qmc.py

```diff
@@ -0,0 +1,621 @@
+import copy
+import numpy as np
+import pytest
+from numpy.testing import (assert_allclose, assert_almost_equal, assert_,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_equal)
+from pytest import raises as assert_raises
+
+from scipy.stats import shapiro
+
+from scipy.stats._sobol import _test_find_index
+from scipy.stats import qmc
+from scipy.stats._qmc import (van_der_corput, n_primes, primes_from_2_to,
+                              _perturb_discrepancy, _update_discrepancy)
+
+
+class TestUtils(object):
+    def test_scale(self):
+        space = [[0, 0], [1, 1], [0.5, 0.5]]
+        corners = np.array([[-2, 0], [6, 5]])
+        out = [[-2, 0], [6, 5], [2, 2.5]]
+
+        scaled_space = qmc.scale(space, bounds=corners)
+
+        assert_allclose(scaled_space, out)
+
+        scaled_back_space = qmc.scale(scaled_space, bounds=corners,
+                                      reverse=True)
+        assert_allclose(scaled_back_space, space)
+
+    def test_discrepancy(self):
+        space_1 = np.array([[1, 3], [2, 6], [3, 2], [4, 5], [5, 1], [6, 4]])
+        space_1 = (2.0 * space_1 - 1.0) / (2.0 * 6.0)
+        space_2 = np.array([[1, 5], [2, 4], [3, 3], [4, 2], [5, 1], [6, 6]])
+        space_2 = (2.0 * space_2 - 1.0) / (2.0 * 6.0)
+
+        # From Fang et al. Design and modeling for computer experiments, 2006
```

Comment:
Great that there's an example in a reference. I can't find this online, though, except for the [Google preview](https://books.google.com/books?hl=en&lr=&id=VGqbyIUVZw8C&oi=fnd&pg=PP1&dq=Fang+et+al.+Design+and+modeling+for+computer+experiments&ots=4_BfFvH-I8&sig=N23Kk4QwXaEuauQnaMjfcJNbOPI#v=onepage&q=Fang%20et%20al.%20Design%20and%20modeling%20for%20computer%20experiments&f=false). Is this on a page there?

Reply:
I see:
![image](https://user-images.githubusercontent.com/6570539/103120371-42e8f200-462c-11eb-8be8-f8adf91c2e9e.png)

Unfortunately, the design it references is not.

Reply:
You have it in page 55 
<img width="651" alt="doe" src="https://user-images.githubusercontent.com/23188539/103130545-639d6000-469d-11eb-9117-70e8de508091.png">

Reply:
I wonder if it shows different previews to different people?
![image](https://user-images.githubusercontent.com/6570539/104789821-e2406700-574a-11eb-9f5c-5552385dc170.png)

But the screenshot is enough. Thanks; looks good.